### DISCUSSION TITLE
How do you configure client/server to use gRPC

### DISCUSSION THREAD

**Discussion by rickfish** [Netflix/conductor#2944] [General]
I am having a tough time trying to figure out how to configure the server to enable gRPC communication as well as how to start a worker polling with gRPC. Has anyone done this? If so, is it done with configuration properties or some other way?

**Comment by rickfish** [comment]
Well I found the 'conductor.grpc-server.enabled' property to allow the GrpServer to be started so I have that working.

I was hoping the client worker framework would also have the ability to set a grpc-enabled flag and the framework would use the grpc clients to do its job but I can't find such a flag. Do I need to roll my own framework if using grpc to poll/update tasks? I imagine it would be fairly straight-forward to add a configuration flag to the conductor-client module to tell TaskPollExecutor to use the grpc TaskClient instead of the http TaskClient depending on the flag but I am sure there are other things that need to be worked through...

**Comment by apanicker-nflx** [comment]
@rickfish The client module currently only supports REST protocol for communication with the server. The `grpc-client` module provides the grpc client stubs for communication but the automators (TaskRunnerConfigurer and TaskPollExecutor) are not available for this yet. Please feel free to submit a PR to generalize these across both mechanisms.
